taichi/backends/cuda/cuda_profiler.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "taichi/program/kernel_profiler.h"
+#include "taichi/backends/cuda/cuda_driver.h"
+#include "taichi/backends/cuda/cuda_context.h"
+#include "cuda_profiler_functions.h"
+
+#include <string>
+#include <stdint.h>
+
+TLANG_NAMESPACE_BEGIN
+
+typedef enum CUDA_KERNEL_PROFILER {
+  CUDA_KERNEL_PROFILER_UNDEF = 0,
+  CUDA_KERNEL_PROFILER_EVENT = 1,
+  CUDA_KERNEL_PROFILER_CUPTI = 2
+} CUDAKernalProfiler;
+
+struct profilerConfig {
+  bool enable = false;
+  CUDAKernalProfiler profiler_type = CUDA_KERNEL_PROFILER_UNDEF;
+  KernelProfilerMode profiling_mode = KernelProfilerMode::disable;
+  uint32_t num_ranges = 1024 * 16;
```

Comment:
?

Reply:
I'll change variable name `num_ranges` in the next PR that uses CUPTI toolkit